### ISSUE TITLE
Edit the code in codon_feature_analysis.R to give the whole window size not only position 0

### ISSUE THREAD

**Issue by Eldonoho99** [riboviz/RiboViz#470]
We would like to edit the code in [codon_feature_analysis.R](https://github.com/riboviz/riboviz/blob/refactor-codon-pairs-436/rscripts/codon_feature_analysis.R) to give the whole window size instead of just position 0 as shown below. 

```


FilterFeatureTableByRelativePosition <-function(feature_table,relative_pos = 0)
{
  # Create a new tibble listing the feature being studied, and the
  # RelCount at position 0, ie RelCount at the feature_of_interest
  feature_table_filter <- feature_table %>% filter(RelPos == 0) %>% dplyr::select(Feature,RelCount)
  feature_table_filter <- arrange(feature_table_filter, desc(RelCount))
  return(feature_table_filter)
}

```

This will help with the statistical analyses we hope to do.

**Comment by Maberuiz** [comment]
I have created the branch full-file-output-470 to work on this

**Comment by Maberuiz** [comment]
Changed the line:
`WriteCodonFeatureAnalysis(output_dir,feature_rel_use_pos0)`
to
`WriteCodonFeatureAnalysis(output_dir,feature_rel_use)`

So now instead of only giving the data of the position 0, the function that creates the final output file (`WriteCodonFeatureAnalysis`) receives all the information from all the positions and creates a table with all the data

**Comment by ewallace** [comment]
Sounds good! Does it need to calculate `feature_rel_use_pos0` still or can those lines also be removed? I think update that, then put a pull request from `full-file-output-470` into `refactor-codon-pairs-436` and ask me to review.

**Comment by Maberuiz** [comment]
After running the line `Rscript rscripts/codon_feature_analysis.R --filter_for_frame 0 -i data/Mok-simYAL5/A.h5 -d Mok-simYAL5 -g data/Mok-simYAL5/Scer_YAL_5genes_w_250utrs.gff3 -a data/yeast_codon_table.tsv --asite_length data/yeast_standard_asite_disp_length.txt --feature data/codon-pairs.tsv -o .` 
With `codon-pairs.tsv` containing: 
Codon pair
CCA TGG
AGA TGG
GTA GTG

The changed output is the following : 
[feature_analysis.txt](https://github.com/riboviz/riboviz/files/9047019/feature_analysis.txt)
(originally a .tsv but changed it to .txt so I was able to upload it)

**Comment by Maberuiz** [comment]
Regarding to your comment @ewallace . `feature_rel_use_pos0` won't be needed anymore. However, maybe it would be a good idea to consider maintaining both files because when, for example, the window size is set to 50, the complete output will get really crowded, and maybe some insights are lost. What do you think about that @Eldonoho99 ?

**Comment by Eldonoho99** [comment]
Yeah I agree. Could get really big quite fast

**Comment by ewallace** [comment]
Thinking... 61 codons * 101 positions = 6161 rows. That's not huge. And it is easy for users to filter down afterwards.

Yes for codon pairs it would get big.

**Comment by Maberuiz** [comment]
> Thinking... 61 codons * 101 positions = 6161 rows. That's not huge. And it is easy for users to filter down afterwards.
> 
> Yes for codon pairs it would get big.

Should maintain both then?

**Comment by ewallace** [comment]
I think take it out. It's literally one line of code to go from the big table to the small.

**Comment by Maberuiz** [comment]
Created pull request #471 to include the mentioned changes in the original branch `refactor-codon-pairs-436`

**Comment by ewallace** [comment]
Fixed by #471
